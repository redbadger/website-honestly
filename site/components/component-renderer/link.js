// @flow
import * as React from 'react';
import styles from './styles.css';

type LinkProps = {
  children?: React.Node,
  href: string
}

export default function Link({ children, href }: LinkProps) {
  return (
    <a className={styles.a} href={href}>
      {children}
    </a>
  );
}
