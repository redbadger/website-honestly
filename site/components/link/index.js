// @flow
import React, { PropTypes } from 'react';
import { NavigationLink } from 'navigation-react';

type LinkProps = {[id: string]: any}
export default function Link(props: LinkProps) {
  const linkProps = Object.assign({}, props, {
    stateKey: props.to,
  });

  return (
    <NavigationLink
      {...linkProps}
    >
      {props.children}
    </NavigationLink>
  );
}

Link.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};
