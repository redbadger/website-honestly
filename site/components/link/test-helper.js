// @flow

import { StateNavigator, HTML5HistoryManager } from 'navigation';
import * as React from 'react';
import { NavigationHandler } from 'navigation-react';
import makeAppNavigator from '../../routes';

export function mockNavigator(parentKey?: string = 'foo') {
  // The constructor is checked in prop-types,
  // so we use it as base and then mutate

  const navigator = new StateNavigator(
    [
      { key: 'foo', route: 'foo' },
      { key: 'bar', route: 'bar' },
      { key: 'barChild', route: 'bar/child', parentKey: 'bar' },
    ],
    new HTML5HistoryManager(),
  );

  navigator.navigate(parentKey);

  return navigator;
}

type Props = {
  stateNavigator?: StateNavigator,
  children: React.Node,
};

const initStateNavigator = () => {
  const navigator = makeAppNavigator();

  navigator.navigateLink('/');

  return navigator;
};

export const Context = (props: Props) => {
  const { children, stateNavigator = initStateNavigator() } = props;

  return <NavigationHandler stateNavigator={stateNavigator}>{children}</NavigationHandler>;
};
