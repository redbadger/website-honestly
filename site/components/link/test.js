// @flow
import React from 'react';
import { shallow } from 'enzyme';
import Link from '.';


describe('components/link', () => {
  it('renders OK children', () => {
    shallow(<Link to="foo">Hello</Link>);
  });
});
