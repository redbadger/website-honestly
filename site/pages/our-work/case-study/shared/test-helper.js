// @no-flow
import React from 'react';

import styles from './what-to-read-next/style.css';
import Link from '../../../../components/link';
import WhatToReadNextSlice from './what-to-read-next/slice';

const MockedWhatToReadNext = () => {
  const nextSlices = [
    {
      name: 'Pride in London',
      tagline: 'Delivering a best-in-class digital experience',
      image: 'pride.gif',
      linkKey: 'prideCaseStudy',
    },
    {
      name: 'Financial Times',
      tagline: 'Lasting change for a media giant',
      image: 'financialTimesReadMore.jpg',
      linkKey: 'financialTimesCaseStudy',
    },
    {
      name: 'Car Trawler',
      tagline: 'Creating complete CMS control for travel brands',
      image: 'carTrawler.jpg',
      linkKey: 'carTrawlerCaseStudy',
    },
  ];
  return (
    <div className={styles.whatNext}>
      <div className={styles.whatNext__tilesContainer}>
        {nextSlices.map((slice, index) => (
          /* disabled because list is static */
          /* eslint-disable react/no-array-index-key */
          <WhatToReadNextSlice
            key={`${slice.name}-${index}`}
            name={slice.name}
            tagline={slice.tagline}
            image={slice.image}
            linkKey={slice.linkKey}
          />
          /* eslint-enable react/no-array-index-key */
        ))}
      </div>
      <Link to="ourWorkPage" className={styles.whatNext__button}>
        See more of our work
      </Link>
    </div>
  );
};

export default MockedWhatToReadNext;
