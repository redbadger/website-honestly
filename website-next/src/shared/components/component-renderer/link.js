import React, { Component } from 'react';
import styles from './styles.css';

export default class Link extends Component {
  static propTypes = {
    children: React.PropTypes.node,
    href: React.PropTypes.string.isRequired,
  };

  render() {
    return (
      <a className={styles.a} href={this.props.href}>{this.props.children}</a>
    );
  }
}
