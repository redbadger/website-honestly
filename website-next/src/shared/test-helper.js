import TestUtils from 'react-addons-test-utils';
import equals from 'lodash/isEqual';
import ShallowTestUtils from 'react-shallow-testutils';

export function render(component) {
  const renderer = TestUtils.createRenderer();
  renderer.render(component);
  return renderer.getRenderOutput();
}

export function hasProp(type, value) {
  return (component) => component.props && component.props[type] === value;
}

export function isEqual(value) {
  return (component) => equals(value, component);
}

export function findOne(predicate, component) {
  const elem = ShallowTestUtils.findAll(component, predicate);

  if (elem.length !== 1) {
    throw new Error('Did not find exactly one match');
  }

  return elem[0];
}

export function containsOne(value, component) {
  return !!findOne(isEqual(value), component);
}

export function findWithType(type, component) {
  return ShallowTestUtils.findWithType(component, type);
}

export function findAllWithType(type, component) {
  return ShallowTestUtils.findAllWithType(component, type);
}
