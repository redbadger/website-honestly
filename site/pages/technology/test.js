import React from 'react';
import { render } from 'enzyme';
import MockDate from 'mockdate';

import { Context } from '../../components/link/test-helper';

import TechnologyPage from './index';

const triedAndTestedBlogPosts = [
  {
    url: 'https://www.somefakebadgersite.com/blog_01',
    title: 'this is a story all about how',
    author: 'The Fresh Prince',
    excerpt: 'my life got flipped turned upside down',
    date: '1988-02-11',
  },
];
const growingTrendsBlogPosts = [
  {
    url: 'https://www.somefakebadgersite.com/blog_02',
    title: 'I would walk 500 miles',
    author: 'The Proclaimers',
    excerpt: 'and I would walk 500 more',
    date: '1986-01-23',
  },
];

describe('site/pages/technology', () => {
  beforeAll(() => {
    MockDate.set('11/05/2019');
  });

  it('renders correctly', () => {
    expect(
      render(
        <Context>
          <TechnologyPage
            triedAndTestedBlogPosts={triedAndTestedBlogPosts}
            growingTrendsBlogPosts={growingTrendsBlogPosts}
          />
        </Context>,
      ),
    ).toMatchSnapshot();
  });
  afterAll(() => {
    MockDate.reset();
  });
});
