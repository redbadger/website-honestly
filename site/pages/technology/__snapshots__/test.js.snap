// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`site/pages/technology renders correctly 1`] = `
Array [
  <section
    class="techSlice "
    data-cy="tech-slice"
  >
    <div
      class="techSliceIntro"
    >
      <h1
        class="heading"
      >
        We love tech. But we only use what’s right for the job.
      </h1>
      <p
        class="projectsBlerb"
      >
        Here is a selection of what we’ve used on recent projects.
      </p>
    </div>
    <ul
      class="techList"
    >
      <li
        class="techListItem"
      >
        <figure
          class="techItem"
        >
          <img
            alt="React + React Native logo"
            class="techLogo"
            src="react_x1.png"
            srcset="react_x1.png 90w, react_x2.png 150w"
          />
          <figcaption
            class="techName"
          >
            React + React Native
          </figcaption>
        </figure>
      </li>
      <li
        class="techListItem"
      >
        <figure
          class="techItem"
        >
          <img
            alt="GraphQL logo"
            class="techLogo"
            src="graphql_x1.png"
            srcset="graphql_x1.png 90w, graphql_x2.png 150w"
          />
          <figcaption
            class="techName"
          >
            GraphQL
          </figcaption>
        </figure>
      </li>
      <li
        class="techListItem"
      >
        <figure
          class="techItem"
        >
          <img
            alt="Java logo"
            class="techLogo"
            src="java_x1.png"
            srcset="java_x1.png 90w, java_x2.png 150w"
          />
          <figcaption
            class="techName"
          >
            Java
          </figcaption>
        </figure>
      </li>
      <li
        class="techListItem"
      >
        <figure
          class="techItem"
        >
          <img
            alt="ESNext logo"
            class="techLogo"
            src="es_x1.png"
            srcset="es_x1.png 90w, es_x2.png 150w"
          />
          <figcaption
            class="techName"
          >
            ESNext
          </figcaption>
        </figure>
        <img
          alt=""
          class="techPlusSymbol"
          src="plus01_x2.png"
        />
      </li>
      <li
        class="techListItem"
      >
        <figure
          class="techItem"
        >
          <img
            alt="Flow + Typescript logo"
            class="techLogo--tall"
            src="flow_x1.png"
            srcset="flow_x1.png 90w, flow_x2.png 150w"
          />
          <figcaption
            class="techName"
          >
            Flow + Typescript
          </figcaption>
        </figure>
        <img
          alt=""
          class="techPlusSymbol"
          src="plus02_x2.png"
        />
      </li>
      <li
        class="techListItem"
      >
        <figure
          class="techItem"
        >
          <img
            alt="NodeJS logo"
            class="techLogo"
            src="nodejs_x1.png"
            srcset="nodejs_x1.png 90w, nodejs_x2.png 150w"
          />
          <figcaption
            class="techName"
          >
            NodeJS
          </figcaption>
        </figure>
      </li>
    </ul>
    <ul
      class="techList"
    >
      <li
        class="techListItem"
      >
        <figure
          class="techItem"
        >
          <img
            alt="AWS logo"
            class="techLogo--wide"
            src="AWS_x1.png"
            srcset="AWS_x1.png 90w, AWS_x2.png 150w"
          />
          <figcaption
            class="techName"
          >
            AWS
          </figcaption>
        </figure>
      </li>
      <li
        class="techListItem"
      >
        <figure
          class="techItem"
        >
          <img
            alt="GCP logo"
            class="techLogo"
            src="GCP_x1.png"
            srcset="GCP_x1.png 90w, GCP_x2.png 150w"
          />
          <figcaption
            class="techName"
          >
            GCP
          </figcaption>
        </figure>
      </li>
      <li
        class="techListItem"
      >
        <figure
          class="techItem"
        >
          <img
            alt="Golang logo"
            class="techLogo--tall"
            src="golang_x1.png"
            srcset="golang_x1.png 90w, golang_x2.png 150w"
          />
          <figcaption
            class="techName"
          >
            Golang
          </figcaption>
        </figure>
      </li>
      <li
        class="techListItem"
      >
        <figure
          class="techItem"
        >
          <img
            alt="Docker logo"
            class="techLogo--wide"
            src="docker_x1.png"
            srcset="docker_x1.png 90w, docker_x2.png 150w"
          />
          <figcaption
            class="techName"
          >
            Docker
          </figcaption>
        </figure>
        <img
          alt=""
          class="techPlusSymbol"
          src="plus03_x2.png"
        />
      </li>
      <li
        class="techListItem"
      >
        <figure
          class="techItem"
        >
          <img
            alt="Kubernetes logo"
            class="techLogo"
            src="kubernetes_x1.png"
            srcset="kubernetes_x1.png 90w, kubernetes_x2.png 150w"
          />
          <figcaption
            class="techName"
          >
            Kubernetes
          </figcaption>
        </figure>
        <img
          alt=""
          class="techPlusSymbol"
          src="plus04_x2.png"
        />
      </li>
      <li
        class="techListItem"
      >
        <figure
          class="techItem"
        >
          <img
            alt="Istio logo"
            class="techLogo"
            src="istio_x1.png"
            srcset="istio_x1.png 90w, istio_x2.png 150w"
          />
          <figcaption
            class="techName"
          >
            Istio
          </figcaption>
        </figure>
      </li>
    </ul>
  </section>,
  <section
    class="yellowSlice"
  >
    <div
      class="container"
    >
      <section
        class="techLabSlice"
      >
        <h2
          class="heading heading__mobile"
        >
          TechLab_Report
          <sup>
            19
          </sup>
        </h2>
        <picture>
          <source
            class="techLabImage"
            srcset="glasses.jpg, glassesX2.jpg 2x, glassesX3.jpg 3x"
          />
          <img
            alt="laboratory safety glasses"
            class="techLabImage"
            src="glasses.jpg"
          />
        </picture>
        <div
          class="headingContainer"
        >
          <h2
            class="heading heading__desktop"
          >
            TechLab_Report
            <sup>
              19
            </sup>
            <br />
            – our view on what matters in tech
          </h2>
          <p>
            Our view on what matters in tech
          </p>
          <a
            class="exploreBtn"
            href="https://hubs.ly/H0gFmpQ0"
          >
            Explore the report
          </a>
        </div>
      </section>
    </div>
  </section>,
  <section
    class="pastAndFuture"
  >
    <div
      class="leftBlogs"
    >
      <section
        class="blogSlice "
      >
        <div
          class="sliceContainer"
        >
          <h2
            class="blogSliceTitle"
          >
            Tried and tested
          </h2>
          <ul
            class="blogLinkList "
          >
            <li>
              <a
                class="link"
                href="https://www.somefakebadgersite.com/blog_01"
              >
                <div
                  class="authorTitle"
                >
                  <span
                    class="linkAuthor "
                  >
                    The Fresh Prince
                  </span>
                   
                  <span
                    class="publishTime"
                  >
                    32 years ago
                  </span>
                </div>
                <div
                  class="linkEntry"
                >
                  <div
                    class="linkTitle "
                  >
                    <h3>
                      this is a story all about how
                    </h3>
                  </div>
                  <p
                    class="excerpt"
                  >
                    my life got flipped turned upside down
                  </p>
                </div>
              </a>
            </li>
          </ul>
        </div>
      </section>
    </div>
    <div
      class="rightBlogs"
    >
      <section
        class="blogSlice blogSliceRight"
      >
        <div
          class="sliceContainer"
        >
          <h2
            class="blogSliceTitle"
          >
            Growing trends
          </h2>
          <ul
            class="blogLinkList blogLinkListRight"
          >
            <li>
              <a
                class="link"
                href="https://www.somefakebadgersite.com/blog_02"
              >
                <div
                  class="authorTitle"
                >
                  <span
                    class="linkAuthor linkAuthorBlack"
                  >
                    The Proclaimers
                  </span>
                   
                  <span
                    class="publishTime"
                  >
                    34 years ago
                  </span>
                </div>
                <div
                  class="linkEntry"
                >
                  <div
                    class="linkTitle linkTitleBlack"
                  >
                    <h3>
                      I would walk 500 miles
                    </h3>
                  </div>
                  <p
                    class="excerpt"
                  >
                    and I would walk 500 more
                  </p>
                </div>
              </a>
            </li>
          </ul>
        </div>
      </section>
    </div>
  </section>,
  <section
    class="webinar"
  >
    <div
      class="webinarInner"
    >
      <h2
        class="webinarText"
      >
        Did you miss our tech webinars on Serverless, GraphQL and more?
      </h2>
      <div
        class="hubspotBtn webinarBtnWrapper"
      >
        <span
          class="hs-cta-wrapper"
          id="hs-cta-wrapper-c87452af-5599-4904-a20c-7d9d48886193"
        >
          <span
            class="hs-cta-node hs-cta-c87452af-5599-4904-a20c-7d9d48886193"
            id="hs-cta-c87452af-5599-4904-a20c-7d9d48886193"
          >
            <a
              href="https://cta-redirect.hubspot.com/cta/redirect/4210858/c87452af-5599-4904-a20c-7d9d48886193"
              target="_blank"
            >
              <img
                alt="Watch them now"
                class="hs-cta-img"
                height="112"
                id="hs-cta-img-c87452af-5599-4904-a20c-7d9d48886193"
                src="https://no-cache.hubspot.com/cta/default/4210858/c87452af-5599-4904-a20c-7d9d48886193.png"
                style="border-width:0px;"
                width="448"
              />
            </a>
          </span>
          <script
            charset="utf-8"
            src="https://js.hscta.net/cta/current.js"
          />
          <script
            type="text/javascript"
          >
             hbspt.cta.load(4210858, 'c87452af-5599-4904-a20c-7d9d48886193', {}); 
          </script>
        </span>
      </div>
    </div>
  </section>,
  <section
    class="contactUsContainer"
    data-cy="contact-us-slice"
    id="contactUs"
  >
    <h2
      class="header"
    >
      We can help you
    </h2>
    <div
      class="contentContainer"
    >
      <ul
        class="list"
      >
        <li
          class="item"
        >
          Navigate the open source revolution
        </li>
        <li
          class="item"
        >
          Choose the right tech for the job
        </li>
        <li
          class="item"
        >
          Adopt meticulous engineering practices
        </li>
        <li
          class="item"
        >
          Enable continuous deployment
        </li>
        <li
          class="item"
        >
          Increase speed to market
        </li>
        <li
          class="item"
        >
          Create value for your customers
        </li>
      </ul>
      <div
        class="imgContainer"
      />
    </div>
    <div>
      <a
        class="mailToLink"
        data-cy="email-link"
        href="mailto:hello@red-badger.com?Subject=Can%20you%20help%20me%20with%20..."
      >
        Send an email
      </a>
      <span
        class="talkToUs"
      >
        hello@red-badger.com
      </span>
    </div>
  </section>,
]
`;
